python/tests/sql/test_dataframe_api.py

```diff
@@ -0,0 +1,1087 @@
+import math
+
+from pyspark.sql import functions as f
+
+from sedona.sql import st_functions as stf
+from sedona.sql import st_constructors as stc
+from sedona.sql import st_aggregates as sta
+from sedona.sql import st_predicates as stp
+from tests.test_base import TestBase
+
+class TestDataFrameAPI(TestBase):
+
+    def test_call_function_using_columns(self):
```

Comment:
Do you think parametrised tests can help to simplify the tests ? I mean https://docs.pytest.org/en/6.2.x/parametrize.html ?

Reply:
Then you can use argument list, function name to apply and also some fixtures to provide dataframe ?

Reply:
Ooohhhh! I have not used parameterized tests before. I'm excited to give them a shot.

Reply:
Alright. I gave this a shot. Let me know if it isn't what you were hoping for.

Also: I don't know what the proper etiquette is on github. Do I click the "Resolve Conversation" button when I think I did it or do the folks requesting code changes do that when they are satisfied?